core/src/main/java/org/apache/calcite/adapter/java/ReflectiveSchema.java

```diff
@@ -75,6 +78,8 @@
     extends AbstractSchema {
   private final Class clazz;
   private final Object target;
+  private Types.FieldsOrdering fieldsOrdering;
+  private @Nullable Map<Class, List<Field>> classFieldsMap;
```

Comment:
Do you think [ClassValue](https://docs.oracle.com/javase/8/docs/api/java/lang/ClassValue.html) would be a better fit?

Reply:
I was not familiar with the `ClassValue` interface, from what I understood by looking it up, it looks beneficial when the `Map` could be the only impediment to the disposal of classes by the GC. 

I think I would anyway need to pass (via the constructor of `MyClassValue`) and store a Map internally to implement the `computeValue` method, so it won’t probably help much. I also expect the list of classes in the map to be of a reasonable size, since it’s coming from the user.

What do you think?